hack/getting_started.sh

```diff
@@ -0,0 +1,591 @@
+    num=`echo "ibase=2;  $1" | bc`
+    returnValue=$(num2IP $num)
+    echo ${returnValue}
+}
+
+function calcControlPlaneIP() {
+    subNet=`docker network inspect kind | jq -r 'map(.IPAM.Config[].Subnet) []'`
+    gateways=`docker network inspect kind | jq -r 'map(.IPAM.Config[].Gateway) []'`
+    occupiedIPList=`docker network inspect kind | jq -r 'map(.Containers[].IPv4Address) []'`
+
+    for line in ${subNet}
+    do
+        echo ${line} | grep ":"
+        if [ $? -ne 0 ]; then
+            IPMask=${line}
+            break
+        fi
+    done
+
+    echo "IPMask is ${subNet}"
+    echo "occupiedIPList:${occupiedIPList}"
+    echo "gateways:${gateways}"
+
+    ip=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\1/p'`
+    ipSubNetBit=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\2/p'`
+    ipSubHostBit=$[32-${ipSubNetBit}]
+
+    IFS=.
+    hexIP=`for str in ${ip}; do printf "%02X" $str; done`
+    binaryIP=`echo "ibase=16; obase=2; ${hexIP}" | bc`
+
+    ipSubNet=`echo ${binaryIP:0:${ipSubNetBit}}`
+
+    full0="00000000000000000000000000000001"
+    full1="11111111111111111111111111111110"
+
+    minIPBinary=${ipSubNet}`echo ${full0:${ipSubNetBit}}`
+    maxIPBinary=${ipSubNet}`echo ${full1:${ipSubNetBit}}`
+
+    minIPint=`echo "ibase=2; ${minIPBinary}"|bc`
+    maxIPint=`echo "ibase=2; ${maxIPBinary}"|bc`
+
+    for((i=${minIPint};i<=${maxIPint};i++));  
+    do
+        ip=$(num2IP ${i})
+
+        isIPOccupied "${ip}" "${occupiedIPList}" "${gateways}"
+        if [ $? == 0 ]; then
+            controlPlaneEndPointIp=${ip}
+            echo "CONTROL_PLANE_ENDPOINT_IP is ${controlPlaneEndPointIp}"
+            return
+        fi
+    done
+
+    echo "Can't get an available IP for control plane endpoint, exit...."
+    exit 1
+
+}
+
+function installByohProvider() {
+    local writeByoh=0
+    local clusterCtlYamlFile="${HOME}/.cluster-api/clusterctl.yaml"
+    local maxRunTimes=40
+    local waitTime=20
+
+    # Write clusterctl.yaml
+    if [ ! -f "${clusterCtlYamlFile}" ]; then
+        writeByoh=1
+        touch ${clusterCtlYamlFile}
+    else
+        grep -q "byoh" ${clusterCtlYamlFile}
+        if [ $? -ne 0 ] ; then
+            writeByoh=1
+        fi
+    fi 
+
+    if [ ${writeByoh} -eq 1 ] ; then
+        cat>>${clusterCtlYamlFile}<<EOF
+providers:
+  - name: byoh
+    url: https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/latest/infrastructure-components.yaml
+    type: InfrastructureProvider
+EOF
+    fi
+
+    clusterctl config repositories | grep -q byoh
+    if [ $? -ne 0 ] ; then
+        echo "Config clusterctl failed..."
+        exit 1
+    fi
+
+    runCmd "clusterctl init --infrastructure byoh" 0 " Transforming the Kubernetes cluster into a management cluster..."
+
+    # Waiting for byoh provider is totally ready
+    for((i=1;i<=${maxRunTimes};i++));  
+    do  
+        byohStatus=$(kubectl get pods --all-namespaces | grep byoh-controller-manager | awk '{print $4}')
```

Comment:
Suggestion: We could provide a specific namespace i.e. `byoh-system`. I was thinking maybe we could use deployment to see `byoh` provider is installed and running as below 
```
kubectl get deployment byoh-controller-manager -n byoh-system | awk '{print $4}'
```

Reply:
It can easy to know the status of pods, if this is "Running", it means it is ok.
$ kubectl get pods --all-namespaces | grep byoh-controller-manager
byoh-system                         byoh-controller-manager-6b59775cfd-2bp92                        2/2     Running   0          26h

It'll be harder to extract the status of deployment, from the view of shell programming.
$ kubectl get deployment byoh-controller-manager -n byoh-system
NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
byoh-controller-manager   1/1     1            1           26h

Reply:
oh sorry, I wasn't making sense in my above comment. What I wanted to say instead is to validate on available > 0(running pods)

`kubectl get deployment byoh-controller-manager -n byoh-system --no-headers | awk '{print $4}'` 
or
`kubectl get deployment byoh-controller-manager -n byoh-system -o json | jq .status.readyReplicas`

These are just some alternatives and I am not suggesting one over the other.

Reply:
@dharmjit Got it, Thanks.